src/components/Modals/FiatRamps/views/Overview.tsx

```diff
@@ -121,17 +139,76 @@ export const Overview: React.FC<OverviewProps> = ({
   chainId,
   setChainId,
   chainAdapterManager,
+  handleAccountIdChange,
+  accountId,
 }) => {
   const translate = useTranslate()
   const { fiatRampAction } = useParams<{ fiatRampAction: FiatRampAction }>()
   const toast = useToast()
   const { fiatRamps } = useModal()
-
-  const [shownOnDisplay, setShownOnDisplay] = useState<Boolean | null>(null)
-
   const {
     state: { wallet },
   } = useWallet()
+  const multiAccountsEnabled = useFeatureFlag('MultiAccounts')
+  const [shownOnDisplay, setShownOnDisplay] = useState<Boolean | null>(null)
+  const accountBalances = useAppSelector(selectPortfolioAccountBalances)
+  const marketData = useAppSelector(state =>
+    selectMarketDataById(state, selectedAsset?.assetId ?? ''),
+  )
+  const [accountAddress, setAccountAddress] = useState<string | null>(null)
+
+  // TODO: change the following with `selectPortfolioAccountMetadataByAccountId`
+  // once web/#2632 got merged
+  const accountsMeta = useAppSelector(selectPortfolioAccountMetadata)
+
+  useEffect(() => {
+    if (!(wallet && accountId && selectedAsset)) return
+    const { chainId: assetChainId } = fromAssetId(selectedAsset.assetId)
+    const { chainId: accountChainId } = fromAccountId(accountId)
+    // race condition between selected asset and selected account
+    if (assetChainId !== accountChainId) return
+    const chainAdapter = chainAdapterManager.get(assetChainId)
+    if (!chainAdapter) return
+    const accountMeta = accountsMeta[accountId]
+    const accountType = accountMeta?.accountType
+    const bip44Params = accountMeta?.bip44Params
+    if (!bip44Params) return
```

Comment:
Add a guard for UTXOs `ChainNamespace` and missing `accountType` - we should now expect an `UtxoAccountType` consistently for UTXOs. 

```suggestion
    if (!bip44Params) return
    
    const { chainNamespace } = fromChainId(accountChainId)
    if (CHAIN_NAMESPACE.Utxo === chainNamespace && !accountType) return
```